src/main/java/cubicchunks/lighting/LightingManager.java

```diff
@@ -46,7 +46,8 @@ public LightingManager(ICubicWorld world) {
 
 		this.skylightCubeDiffuseProcessor = new SkyLightCubeDiffuseProcessor(world, "Sky Light Diffuse", 5);
 
-		this.firstLightProcessor = new FirstLightProcessor("First Light", world.getCubeCache(), 1);
+		// TODO: Setting GeneratorStage to null. Analyze effects.
+		this.firstLightProcessor = new FirstLightProcessor(null, "First Light", world.getCubeCache(), 1);
```

Comment:
So you want to start lighting not exactly at stage x, but any time after stage x correct? If this is the case, I understand what you meant by non-linear.